### PR TITLE
Add MongoDB Atlas terraform module

### PR DESCRIPTION
**Pull request by gustavoborragan**

### CODE REVIEW

File: modules/mongodb-atlas/README.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+| enabled_datadog_integration | Whether or not the Datadog integration is enabled | `bool` | `false` | no |
+| datadog_api_key_name | The Datadog API key name | `string` | `""` | no |
+
+### Set a data .tfvars
+
+```hcl
+# Global variables
+mongo_region               = "EUROPE_WEST"
+provider_name              = "AZURE"
+global_resource_group_name = "example"
```
Comment (alambike) [inline modules/mongodb-atlas/README.md:135]: What is the global_resource_group_name? 
What provider_names are available?

File: modules/mongodb-atlas/README.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+# Endpoint section variables
+subnet_name                              = "example-subnet"
+vnet_name                                = "example-vnet"
+vnet_resource_group_name                 = "example-vnet-rg"
+endpoint_name                            = "example-endpoint"
+endpoint_location                        = "westeurope"
+endpoint_resource_group_name             = "example-endpoint-rg"
+endpoint_connection_is_manual_connection = true
+endpoint_connection_request_message      = "Example request message"
```
Comment (alambike) [inline modules/mongodb-atlas/README.md:221]: This parameters are exclusive to Azure right? Maybe should be in a specific provider block?
```
provider:
  name: "AZURE"
  disk_type: xxx
  instance_size_name: xxxx
  azure_private_link:
     private link parameters?
```

File: modules/mongodb-atlas/README.md (hunk 1 above)
Comment (gustavoborragan) [reply modules/mongodb-atlas/README.md:135]: Added more info

**Review by jcframil — APPROVED** [review]
lgtm

File: modules/mongodb-atlas/README.md (hunk 2 above)
Comment (gustavoborragan) [reply modules/mongodb-atlas/README.md]: Set a new variable:

```hcl
provider = {
  provider_name                              = "AZURE"
  global_resource_group_name                 = "example"
  cluster_provider_disk_type_name            = "P6"
  cluster_provider_instance_size_name        = "M10"
  network = {
    subnet_name                                = "example-subnet"
    vnet_name                                  = "example-vnet"
    vnet_resource_group_name                   = "example-vnet-rg"
    endpoint_name                              = "example-endpoint"
    endpoint_location                          = "westeurope"
    endpoint_resource_group_name               = "example-endpoint-rg"
    endpoint_connection_is_manual_connection   = true
    endpoint_connection_request_message        = "Example request message"
  }
}
```

File: modules/mongodb-atlas/datadog_integration.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Datadog API Key section
+# https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/api_key
+resource "datadog_api_key" "datadog_api_key" {
+  name = var.datadog_api_key_name
+}
```
Comment (alambike) [inline modules/mongodb-atlas/datadog_integration.tf:5]: This resource should have count conditional right?

File: modules/mongodb-atlas/datadog_integration.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  count = var.enabled_datadog_integration ? 1 : 0
+  name  = var.datadog_api_key_name
+
+  # Ensure data source is fetched after the API key is created
+  depends_on = [datadog_api_key.datadog_api_key]
+}
+
+# Store the Datadog API key in Azure Key Vault
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_secret
+resource "azurerm_key_vault_secret" "datadog_api_key_secrets" {
```
Comment (alambike) [inline modules/mongodb-atlas/datadog_integration.tf:19]: The usage of the azure_keyvault should be optional too, if you're in AWS you will not use Azure Keyvault. Maybe is better to have this as a sensitive output, and have another resource (and claim?) to upload it to the vault, whatever vault system you want to use in your organization

File: modules/mongodb-atlas/network.tf
Comment (alambike) [inline modules/mongodb-atlas/network.tf:1]: I would rename this file to `azure_privatelink.tf`

File: modules/mongodb-atlas/network.tf [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+  # Ignore changes to tags
+  lifecycle {
+    ignore_changes = [tags]
+  }
+}
+
+# MongoDB Atlas Private Link Endpoint Service resource
+# https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint_service
+resource "mongodbatlas_privatelink_endpoint_service" "privatelink_endpoint_service" {
```
Comment (alambike) [inline modules/mongodb-atlas/network.tf:64]:
```suggestion
resource "mongodbatlas_privatelink_endpoint_service" "this" {
```

File: modules/mongodb-atlas/network.tf [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+data "azurerm_subnet" "subnet" {
+  count                = var.provider.provider_name == "AZURE" ? 1 : 0
+  name                 = var.provider.network.subnet_name
+  virtual_network_name = var.provider.network.vnet_name
+  resource_group_name  = var.provider.network.vnet_resource_group_name != "" ? var.provider.network.vnet_resource_group_name : var.provider.global_resource_group_name
+}
+
+# Azure Private Endpoint resource
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/private_endpoint
+resource "azurerm_private_endpoint" "private_endpoint" {
```
Comment (alambike) [inline modules/mongodb-atlas/network.tf:38]:
```suggestion
resource "azurerm_private_endpoint" "that" {
```

**Pull request closed without merging**